sherpa/stats/tests/test_stats.py

```diff
@@ -299,6 +299,15 @@ def test_wstat_error(self):
         fit = Fit(data, self.model, WStat(), NelderMead())
         self.assertRaises(StatErr, fit.fit)
 
+    def test_chi2datavar(self):
+        num = 3
+        xy = numpy.array(range(num))
+        ui.load_arrays(1, xy, xy, Data1D)
+        ui.set_stat('chi2datavar')
+        err = ui.get_staterror()
+        for index, yyy in enumerate(xy):
```

Comment:
Numpy includes several routines for testing whether arrays match, within a tolerance, which would avoid the explicit looping (actually, `numpy.allclose` also does so as well, but it just returns `True` or `False` and the testing routines work with `unittest`). The routines are in [`numpy.testing`](http://docs.scipy.org/doc/numpy/reference/routines.testing.html) and probably the one wanted here would be [`numpy.testing.assert_allclose`](http://docs.scipy.org/doc/numpy/reference/generated/numpy.testing.assert_allclose.html#numpy.testing.assert_allclose), which should let you replace the for loop by (untested, assuming the symbol has been explicitly imported)

    assert_allclose(err, numpy.sqrt(xy), rtol=1e-7, atol=1e-7)

It also provides a more-useful error message than the `assert` statement used on line 309.